src/azure-cli/azure/cli/command_modules/appconfig/_utils.py

```diff
@@ -254,3 +261,73 @@ def parse_tags_to_dict(tags):
                 tags_dict[tag_key] = tag_value
         return tags_dict
     return tags
+
+
+def strip_json_comments(json_string):
+    current_index = 0
+    length = len(json_string)
+    result = []
+
+    def __is_escaped(json_string, char_index):
```

Comment:
If we have these as private is there a good reason to define them in the method, it does make it a bit of a challenge to read.

Reply:
We can define it outside the method. I just felt passing the length of the string around could be avoided. For readability, I can move them into their own method.

Reply:
Updated to have separate methods outside the main method